application_test.go

```diff
@@ -427,3 +427,20 @@ func TestApplicationConfiguration(t *testing.T) {
 	assert.True(t, ok)
 	assert.Equal(t, ErrCodeNotFound, apiErr.ErrCode)
 }
+
+func TestWaitOnApplication(t *testing.T) {
+	endpoint := newFakeMarathonEndpoint(t, nil)
+	defer endpoint.Close()
+	client := endpoint.Client.(*marathonClient)
+
+	c := make(chan bool)
+	err := client.waitOnApplication(fakeAppName, c)
+	assert.NoError(t, err)
+
+	c = make(chan bool)
+	go func() {
+		c <- true
+	}()
```

Comment:
That's pretty good already. I think we need to do one last thing though:

Unless I am mistaken, we still cannot be certain that we will make it into one iteration of `waitOnApplication`'s for-loop before the stop signal arrives. How about sleeping half a second inside this go function to give the the system-under-test enough time to hit a 404?